Jellyfin.Server/Migrations.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Collections.Generic;
+using MediaBrowser.Common.Configuration;
+using MediaBrowser.Model.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Jellyfin.Server
+{
+    /// <summary>
+    /// The class that knows how migrate between different Jellyfin versions.
+    /// </summary>
+    internal static class Migrations
+    {
+        private static readonly IUpdater[] _migrations =
+        {
+            new Pre10_5()
+        };
+
+        /// <summary>
+        /// Interface that descibes a migration routine.
+        /// </summary>
+        private interface IUpdater
+        {
+            /// <summary>
+            /// Gets maximum version this Updater applies to.
+            /// If current version is greater or equal to it, skip the updater.
+            /// </summary>
+            public abstract Version Maximum { get; }
+
+            /// <summary>
+            /// Execute the migration from version "from".
+            /// </summary>
+            /// <param name="host">Host that hosts current version.</param>
+            /// <param name="logger">Host logger.</param>
+            /// <param name="from">Version to migrate from.</param>
+            /// <returns>Whether configuration was changed.</returns>
+            public abstract bool Perform(CoreAppHost host, ILogger logger, Version from);
+        }
+
+        /// <summary>
+        /// Run all needed migrations.
+        /// </summary>
+        /// <param name="host">CoreAppHost that hosts current version.</param>
+        /// <param name="logger">AppHost logger.</param>
+        /// <returns>Whether anything was changed.</returns>
+        public static bool Run(CoreAppHost host, ILogger logger)
+        {
+            bool updated = false;
+            var version = host.ServerConfigurationManager.CommonConfiguration.PreviousVersion;
+
+            for (var i = 0; i < _migrations.Length; i++)
+            {
+                var updater = _migrations[i];
+                if (version.CompareTo(updater.Maximum) >= 0)
```

Comment:
I'm wondering if it would be better to take an approach similar to how database migrations are done and track whether each migration individually has been applied.  This approach leaves the opportunity for migrations to be applied multiple times accidentally.

This approach could also be annoying for development since migrations can only be applied with a version bump of the application. If I'm running nightlies or running from source on master, I would want the migrations to be applied to my setup as soon as they're merged into master, not wait months until the next version is released.